modules/SofaSparseSolver/src/SofaSparseSolver/SparseCholeskySolver.cpp

```diff
@@ -108,10 +110,117 @@ void SparseCholeskySolver<TMatrix,TVector>::invert(Matrix& M)
     A.nz = -1;							// # of entries in triplet matrix, -1 for compressed-col
     cs_dropzeros( &A );
     tmp.resize(A.n);
-    S = cs_schol (&A, order) ;		/* ordering and symbolic analysis */
+    
+    perm = new int[A.n ];
+    iperm = new int[A.n ];
+
+    fill_reducing_perm(A , perm, iperm); // compute the fill reducing permutation
+    S = symbolic_Chol( &A ); // symbolic analysis
+    permuted_A = cs_permute( &(A), iperm, perm, 1);
+
+    sofa::helper::AdvancedTimer::stepBegin("factorization");
     N = cs_chol (&A, S) ;		/* numeric Cholesky factorization */
+    sofa::helper::AdvancedTimer::stepEnd("factorization");
+}
+
+template<class TMatrix, class TVector>
+void SparseCholeskySolver<TMatrix,TVector>::fill_reducing_perm(cs A,int * perm,int * invperm)
+{
+    int n = A.n;
+    if(d_applyPermutation.getValue() )
+    {
+        int *M_colptr=A.p, *M_rowind=A.i ;
```

Comment:
If this is the same algorithm in LDL, LU, and Cholesky, I suggest to factorize this code. Put this code into a free function, in a file that you would include in LDL, LU and Cholesky.

Reply:
I tough about doing exactly that but didn't know where in the inheritance I should put it.